### PR TITLE
yum: implement feature installroot, issue #1558

### PR DESCRIPTION
**Pull request by verm666**

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @verm666. Since yum is a pretty core feature, might take us a little while to review this one.

**Comment by verm666** [comment]
Ok @gregdek , to be honest current implementation of yum module not good enough and I want to make a refactoring, e.g. for fix issue #948, #1441 and #1558 (this one). Can I show refactored version of yum.py to somebody (it will be more productive then reviewing every PR one by one, I guess)? And more over, in my mind, we should add unit tests for some core modules like this one, apt, etc. What do you think about it?

**Comment by bcoca** [comment]
cc @abadger , verm666 the yum module has many issues, abadger has actually been working with several people to fix them, he is the best person to consult for that.

**Comment by abadger** [comment]
@verm666 Separate PRs are likely better as it makes it quicker and easier to review.  If you want to do a major refactor/restructure we might want to have one PR that just does the refactoring without attempting to add new features.

**Comment by abadger** [comment]
@verm666 Also regarding testing -- unittests are a bit hard as the modules aren't just regular python modules (at least, at this point).  So some framework would have to be built up to make that easier to work with.

We do have integration tests that cover yum.  You can take a look at them and add new tests if you'd like:
https://github.com/ansible/ansible/tree/devel/test/integration/roles/test_yum/tasks

File: packaging/os/yum.py [hunk 1]
```diff
@@ -118,12 +118,22 @@
...
+    aliases: []
+
 notes: []
 # informational: requirements for nodes
 requirements: [ yum ]
-author: 
+author:
     - "Ansible Core Team"
     - "Seth Vidal"
+    - "Eduard Snesarev"
```
Comment (abadger) [inline packaging/os/yum.py:136]: Add your github id here please.

**Comment by abadger** [comment]
- For the centos6 notes, is that just a problem when installroot=/ ?
- We probably need at least one smoketest that shows that we don't totally break installroot functionality in future updates.  More would be better since we go down multiple code paths (using YumBase, using repoquery, and using the yum commandline) in this code.

Code here looks good, though.  So once at least one test is submitted I think we can merge this.

**Comment by verm666** [comment]
> For the centos6 notes, is that just a problem when installroot=/ ?
> Yes

**Comment by abadger** [comment]
Cool.  When you make a PR for testing, just link it here and then we can merge this. :-)

**Comment by gregdek** [comment]
Hmm... has something changed underneath? Looks like this needs rebasing now...

**Comment by verm666** [comment]
Guys, please tell me how I can run tests for yum module on my own copy of ansible. Have you got a docs about it?

**Comment by gregdek** [comment]
OK, so a few issues around this one.

Issue 1: yum is very critical for core, and it's unlikely that we'll be able to do further review on it by the time 2.0 ships, so tagging it with the "next" milestone tag.

Issue 2: @abadger has asked for tests, and @verm666 needs some guidelines for how to do that; do we have anything for him?

Issue 3: this code now needs to be rebased.

**Comment by gregdek** [comment]
@verm666 A friendly reminder: this pull request has been marked as needing your action, and it hasn't been updated in the last month. 

If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we'll keep it open pending your changes.

If we receive no indication from you within a week, we will close this pull request.

**Comment by abadger** [comment]
Oops, sorry I didn't see the question about how to test.

Follow the instructions on how to run from source:
http://docs.ansible.com/ansible/intro_installation.html#running-from-source

Note that you'll need to become root prior to sourcing env-setup so that
the environment variables are set for the root user.

Change to the tests/integration directory.

Run TEST_FLAGS="--tags test_yum" make destructive

That should run the tests in the yum role (it's in the destructive target
because it installs and removes packages on the system.)

I'm on irc.freenode.net (abadger1999 on#ansible-devel) if you ever need to
ping me to get help out remind me that I'm reviewing our meeting a pr of
yours.

**Comment by robynbergeron** [comment]
@verm666 Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If we don't hear from you within another 14 days, we will close this pull request.

**Comment by gregdek** [comment]
Closed due to contributor inactivity.

**Pull request closed without merging**